cirq/pasqal/pasqal_device.py

```diff
@@ -11,52 +11,45 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Any, Dict, FrozenSet, Iterable
+from typing import FrozenSet, Callable, List, Sequence, Any, Union, Dict
+import numpy as np
 from numpy import sqrt
 
 import cirq
+from cirq import GridQubit, LineQubit
+from cirq.ops import NamedQubit
 
-from cirq.pasqal import ThreeDGridQubit
+from cirq.pasqal import ThreeDQubit, TwoDQubit
 
 
 @cirq.value.value_equality
 class PasqalDevice(cirq.devices.Device):
-    """A Pasqal Device with qubits placed on a 3D grid."""
+    """A generic Pasqal device."""
 
-    def __init__(self, control_radius: float,
-                 qubits: Iterable[ThreeDGridQubit]) -> None:
-        """ Initializes the description of the device.
+    def __init__(self, qubits: Sequence[cirq.ops.Qid]) -> None:
+        """Initializes a device with some qubits.
 
         Args:
-            control_radius: the maximum distance between qubits for a controlled
-                gate. Distance is measured in units of the indices passed into
-                the qubit constructor.
-            qubits: Qubits on the device, identified by their x, y, z position.
-                Must be of type ThreeDGridQubit.
-
+            qubits (NamedQubit): Qubits on the device, exclusively unrelated to
```

Comment:
Why does it needs to be `NamedQubit`? If this is a recommendation, explain in the doc. If it's a requirement, it should be represented in the type signature.

Reply:
I see the issue here, we can't represent it easily in the type signature.

Reply:
`NamedQubit`was the only qubit native to Cirq (`Qid`being to general for practical purposes) that would not misleadingly make the user position the qubit in space. We could have created another qubit class specifically for this, but it seemed overkill. I realise this is not optimal, but it felt better than accepting (necessarily) spatially positioned qubits when those positions will ultimately be ignored, or cluttering the module with yet another qubit type. Do you agree?

To be clear, we are requiring the qubits to be `NamedQubit`, so I will represent this in the type signature.

Reply:
> I see the issue here, we can't represent it easily in the type signature.

You're right, I remember now why I couldn't fix the type to restrict the input to NamedQubit. I'm still checking the type upon initialization, so I hope that suffices.